### PR TITLE
#38 sub-workflow for pangenome analysis

### PR DESCRIPTION
**Pull request by jonathho**
- created sub-workflow for pangenome analysis
- manually added PEPPAN modules from #40 for easier merging
- completes #38

### CODE REVIEW

File: main.nf
Comment (anwarMZ) [inline main.nf:1]: This should not be part of the pull request. 

@jonathho Can you please remove this from the PR

File: modules/local/PEPPAN/environment.yml
Comment (anwarMZ) [inline modules/local/PEPPAN/environment.yml:1]: @jonathho & @azmigueldario This environment file contains platform-specific build constraints, that will restrict it from creating environments across platforms. The YAML file for the environment should have minimal packages required and without platform-specific builds. 
For now, maybe remove this from the PR.

File: modules/local/PEPPAN/main.nf [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    tag "$meta.id"
+    label 'process_medium'
+    conda '$moduleDir/environment.yml'
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'docker://cidgoh/peppan:latest' :
+        'cidgoh/peppan:latest' }"
+
+    input:
+    tuple val(meta), path(gff)
+    path (reference_gff)
```
Comment (anwarMZ) [inline modules/local/PEPPAN/main.nf:11]: `path reference_gff`

File: modules/local/PEPPAN/main.nf [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    path "versions.yml"                             ,             emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def gff = params.reference_gff ? "-P params.reference_gff" : ''
+    def version = "1.0.5" // ------> update manually
```
Comment (anwarMZ) [inline modules/local/PEPPAN/main.nf:24]: this should not be done manually. Given that the peppan is installed in the docker we can find out the version.

See example - https://github.com/cidgoh/nf-seqqc/blob/4b83e5e05c042e0352167f3ef915f8ce444aadb3/modules/nf-core/fastp/main.nf#L99

**Review by anwarMZ — CHANGES_REQUESTED** [review]
Some changes are required before I can test it .. I have highlighted them in each file below.

File: modules/local/PEPPAN/main.nf (hunk 1 above)
Comment (jonathho) [reply modules/local/PEPPAN/main.nf:11]: Sorry, I'm not sure I follow this one. Do you mean you would like the line to be like this instead:
`tuple val(meta), path(gff), path(reference_gff)`

Comment (anwarMZ) [reply modules/local/PEPPAN/main.nf:11]: sorry for the confusion. No I was just referring to changing line 11 to `path reference_gff` i-e removing the parenthesis around the variable. It doesn't change anything but I realized that it is inconsistent across the workflow. I will go through this across modules once we have merged these changes.

**Comment by jonathho** [comment]
Sorry for the delay @anwarMZ. I just pushed changes that should resolve the requested changes expect for getting the peppan version. Peppan doesn't have a `--version` option so I was looking for another way to get it. I tried retrieving from line 6 in the dockerfile: `LABEL image.peppan="1.0.5"` but have been unsuccessful so far.

**Review by anwarMZ — APPROVED** [review]
